### DISCUSSION TITLE
Vercel serverless crashed

### DISCUSSION THREAD

**Discussion by andregosling** [vercel/community#888] [Help] [answered]
Hi! I've created a vercel serverless function (not Next.js), to make a url shortener. This service basically make a request, wrapped by a try/catch, and without await operator, to make a fast redirect. But, for sometimes, i receive the crash message 'This Serverless Function has crashed'. Checking the logs i got this, but that error is not clear to me, because there isn't any favicon in my app.

![image](https://user-images.githubusercontent.com/38388768/191748478-1fbfe600-56a8-406a-bd0b-6029334ca79f.png)

I'm also using rewrites at vercel.json 
![image](https://user-images.githubusercontent.com/38388768/191750992-fcc97c1a-974b-4844-8241-b14b0e2f7a58.png)

**Comment by amyegan** [comment, marked as answer]
Hi @andregosling! Can you share the deployment ID so I can see the logs? A [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo, if public) would also be helpful for debugging.

**Reply by andregosling** [reply]
Hi! yeah, for sure. it is F1oVoHoQQuXsd7uToMkgrrYssKLZ. Here also goes a screenshot of a moment with the error was happen. I guess a possible solution is avoid the rewrite to happen if the path is a file.. you know with is the best approach to do this?

**Note**: A didn't have access the path '/favicon.ico', but a token

![image](https://user-images.githubusercontent.com/38388768/191818638-3725d836-dede-4da3-8f5c-ba2dbd4d043c.png)

**Reply by amyegan** [reply]
Thanks for sharing the deployment ID. I'm not sure the favicon is the whole problem. It looks like that error would only interfere with loading a favicon into the browser tab, but not with the rest of your site functionality.

The home page 404 is coming in before the favicon 500 error. 
<img width="880" alt="image" src="https://user-images.githubusercontent.com/8247105/191824559-d0f34150-90a6-4d74-b023-9258bc635981.png">

If you can't share a reproduction in code, can you walk me through how your app _should_ be working and what errors your seeing along the way? I want to make sure I'm not misunderstanding the problem. Links to preview deployment addresses where I can see the intended and unintended behavior will help me get a better view of what is happening.

**Reply by andregosling** [reply]
Yeah, for sure. That works as a simple url redirect. We receive the request at the shortener (eg. `url-shortener.com/:token`), and the handler make a simple POST to my API, to a counter to some the view, and redirect to my main domain (main-domain/path/:token). Like a said, a make this request as sync to not delay the response of the redirect. Almost all the times a access this path at my shortener, all works fine as should, but, for some times, i receive the crash error. I've said about the favicon because all the times that error happens, checking the function logs, the path of the log is `url-shortener.com/favicon.ico`, even i was accessed a token with has nothing to do with the path 'favicon.ico'. It looks like a conflict on my first rewrite, because the public files are computed as a token. 

I'm using the rewrites because i don't wanna to the user access the path 'api', but directly at the root.

**Reply by amyegan** [reply]
You are absolutely right! I finally found where it is happening in the logs. It's treating `favicon.ico` as a token for the redirect to your api path.

<img width="262" alt="image" src="https://user-images.githubusercontent.com/8247105/191835494-be9832fc-515a-4591-b6b3-14bec6742cb6.png">

There are a few of options to handle this:
- Add a generic favicon since "precedence is given to the filesystem prior to rewrites being applied" ([rewrites doc](https://vercel.com/docs/project-configuration#project-configuration/rewrites))
- Catch "favicon.ico" in your function and redirect from there to prevent the error
- Add an entry to handle `/favicon.ico` differently from the rest of your `/:token` sources

**Reply by andregosling** [reply]
Thanks! As create a generic favicon, you mean create a favicon.ico inside my public folder? I was thinking about that before, but is only the favicon.ico which can throw this problem? There isn't another files with i can receive the same bad result? Like robots.txt, or some other files with usually stays in the public folder

**Reply by amyegan** [reply]
Theoretically yes others could have the same result, but this error is only happening because the favicon file doesn't actually exist. The redirect is respecting existing files in your project, but it is redirecting `/favicon.ico` to `/api/favicon.ico` because that file doesn't currently exist in your project. If you actually have robots.txt or other files then they wouldn't be redirected to `/api`

**Reply by andregosling** [reply]
Okay, i see. Thank you very much Amy!
